g2p/mappings/langs/mic/mic_to_ipa.json

```diff
@@ -0,0 +1,154 @@
+[
+   {
+      "in": "\uA78C",
+      "out": "'"
+   },
+   {
+      "in": "í|i'",
+      "out": "iː"
+   },
+   {
+      "in": "á|a'",
+      "out": "aː"
+   },
+   {
+      "in": "é|e'",
+      "out": "eː"
+   },
+   {
+      "in": "ó|o'",
+      "out": "oː"
+   },
+   {
+      "in": "ú|u'",
+      "out": "uː"
+   },
+   {
+      "in": "j",
+      "out": "tʃ"
+   },
+   {
+      "in": "q",
+      "out": "x"
+   },
+   {
+      "in": "y",
+      "out": "j"
+   },
+   {
+      "in": "p",
+      "out": "b",
+      "context_before": "SONORANT",
```

Comment:
I believe, with the way `context_before` is implemented, this context will be matched even if `ː` appears in isolation. I.e., `bːp` would get mapped to `bːb`. I think that's OK, because that's certainly not proper Mi'kmaq input or ipa intermediate, but it's best to be aware of it.

Reply:
> Quickly done!
> 
> See various comments attached to the code.
> 
> More global comment:
> 
> I think there's a problem with all the ^C->eC followed C->Ce with context_after=CONSONANT in both cases, which I think will make ^C end up as eCe (read all these e's as schwas). Maybe instead, C->Ce followed by ^Ce->eC?
> 
> And we have the usual problem with ^, which matches the beginning of the line, not the beginning of the word... Maybe we need to change `g2p convert` to tokenize on spaces before applying the mapping, then `^` and `$` would match on all word boundaries, even within a line.

The point about ^C->əC followed C->Cə is good except the latter rule is only turning C->Cə when followed by a word-final consonant, so it currently predicts sq -> əsəx (after spirantization of the /q/). I suppose you're saying it might be səx, but I'm not actually sure. səx, əsx and əsəx all seem plausible. I've re-ordered the rules as you propose though and I suppose I'll keep my ears out for these kinds of words!

The way `context_before` is implemented is to make a lookbehind by joining all of the characters in the SONORANT variable. The fully expanded regex for the rule above looks like this: `((?<=iː|aː|eː|oː|uː)|(?<=i|a|e|u|o|y|m|n|l))p(?=i|a|e|u|o|iː|aː|eː|oː|uː|y|m|n|l)` the single and two-character sequences are automatically separated into different groups because lookbehinds have to be fixed width. Anyways, long story short, `bːp` won't map to `bːb`.